app/scripts/controllers/buildConfig.js

```diff
@@ -74,6 +109,7 @@ angular.module('openshiftConsole')
                 };
               }
               $scope.buildConfig = buildConfig;
+              $scope.updatedBuildConfig = angular.copy(buildConfig);
```

Comment:
Yeah, easy would be a banner "This resource has been updated. refresh?" and let the user decide if it should replace what they have.  Harder is the diff...

@rhamilto just to keep you in the loop on this chatter :)